ledger/internal/apptxn_test.go

```diff
@@ -3058,3 +3058,122 @@ check:
 	txns(t, l, eval, &fundA, &callA)
 	endBlock(t, l, eval)
 }
+
+func TestForeignAppAccountsAccessible(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	genBalances, addrs, _ := ledgertesting.NewTestGenesis()
+	l := newTestLedger(t, genBalances)
+	defer l.Close()
```

Comment:
I'd prefer you use the testConsensusRange system.  That will allow you to conveniently have tests that show it is not available in old versions, and is available in new ones, including future ones.